content/client-server-api/modules/secrets.md

```diff
@@ -147,58 +203,35 @@ HMAC-SHA-256. The secret is encrypted as follows:
 1.  Given the secret storage key, generate 64 bytes by performing an
     HKDF with SHA-256 as the hash, a salt of 32 bytes of 0, and with the
     secret name as the info. The first 32 bytes are used as the AES key,
-    and the next 32 bytes are used as the MAC key
+    and the next 32 bytes are used as the MAC key.
+
 2.  Generate 16 random bytes, set bit 63 to 0 (in order to work around
     differences in AES-CTR implementations), and use this as the AES
-    initialization vector. This becomes the `iv` property, encoded using
-    base64.
-3.  Encrypt the data using AES-CTR-256 using the AES key generated
-    above. This encrypted data, encoded using base64, becomes the
-    `ciphertext` property.
-4.  Pass the raw encrypted data (prior to base64 encoding) through
-    HMAC-SHA-256 using the MAC key generated above. The resulting MAC is
-    base64-encoded and becomes the `mac` property.
+    initialization vector (IV).
 
-`AesHmacSha2EncryptedData`
+3.  Encrypt the data using AES-CTR-256 using the AES key and IV generated
+    above.
 
-| Parameter  | Type    | Description
-|------------|---------|------------------------------------------------------------------------|
-| iv         | string  |   **Required.** The 16-byte initialization vector, encoded as base64.  |
-| ciphertext | string  |   **Required.** The AES-CTR-encrypted data, encoded as base64.         |
-| mac        | string  |   **Required.** The MAC, encoded as base64.                            |
+4.  Pass the raw encrypted data through HMAC-SHA-256 using the MAC key
+    generated above.
 
-For the purposes of allowing clients to check whether a user has
-correctly entered the key, clients should:
+5.  Encode the IV from step 2, the ciphertext from step 3, and MAC from step 4,
+    using base64, and store them as the `iv`, `ciphertext`, and `mac`
```

Comment:
Question? Padded or unpadded base64? If padded, which variant?

Existing text just says "base64", which implies padded (of some variant). Web uses (I think) padded [RFC4648](https://datatracker.ietf.org/doc/html/rfc4648) base64. But the spec also [says](https://spec.matrix.org/v1.8/client-server-api/#secret-storage)

> it’s expected that most encryption schemes would have `ciphertext` and `ma`c properties, where the `ciphertext` property is the **unpadded** base64-encoded ciphertext

Reply:
Yeah, I made a mistake not specifying this originally.  We want it to be unpadded, to be consistent with everything else in Matrix, but since it was left unspecified, clients should be able to handle either one, for example by base64-decoding the MAC and comparing the binary value instead of comparing the base64 strings.